app/controllers/api/v2/hosts_controller.rb

```diff
@@ -353,6 +341,13 @@ def templates
         end
       end
 
+      api :GET, "/hosts/:id/virtualization", N_("Get virtualization information for the host")
+      param :id, :identifier_dottable, :required => true
+      def virtualization
+        provider = @host.compute_resource.provider.downcase
```

Comment:
`provider` is not always defined in my env and causes the request to fail 
```
Backtrace for 'Action failed' error (NoMethodError): undefined method `provider' for nil:NilClass
```

Reply:
That's because `compute_resource` isn't always defined. I wonder if in that case we should return `404`, since there is no virtualization for the host.